api/ruxos_posix_api/src/imp/time.rs

```diff
@@ -74,6 +74,50 @@ pub unsafe fn sys_clock_settime(_clk: ctypes::clockid_t, ts: *const ctypes::time
     })
 }
 
+/// Sleep until some nanoseconds
+///
+/// TODO: should be woken by signals, and set errno
+/// TODO: deal with flags
+pub unsafe fn sys_clock_nanosleep(
+    which_clock: ctypes::clockid_t,
+    flags: c_int,
+    req: *const ctypes::timespec,
+    rem: *mut ctypes::timespec,
+) -> c_int {
+    syscall_body!(sys_clock_nanosleep, {
+        unsafe {
+            if req.is_null() || (*req).tv_nsec < 0 || (*req).tv_nsec > 999999999 {
```

Comment:
It would be better to replace `999999999` with `const NANO_PER_SECOND = 1000000000` .

Reply:
and leave a comment here: )